worlds/dark_souls_3/__init__.py

```diff
@@ -489,10 +491,11 @@ def _fill_local_item(
         if not candidate_locations:
             warning(f"Couldn't place \"{name}\" in a valid location for {self.player_name}. Adding it to starting inventory instead.")
             location = next(
-                (location for location in self.multiworld.get_locations(self.player) if location.data.default_item_name == item.name),
+                (location for location in cast(List[DarkSouls3Location], self.multiworld.get_locations(self.player)) if location.data.default_item_name == item.name),
```

Comment:
Since this cast comes up multiple times, let's add a `self._get_our_locations()` helper.

Reply:
Fixed for the cases that used `get_locations` but not the filled/unfilled versions which are each only used once